### PR TITLE
feat(ERTP): Make recoverySets optional

### PR DESCRIPTION
**Pull request by erights**
TODO explain

Staged on https://github.com/Agoric/agoric-sdk/pull/8414

### CODE REVIEW

File: packages/ERTP/src/types-ambient.js [hunk 1]
```diff
@@ -132,62 +132,17 @@
...
-
-/**
- * @callback IssuerSplitMany Split a single payment into many payments,
- *   according to the amounts passed in.
- *
- *   If the payment is a promise, the operation will proceed upon resolution.
- * @param {ERef<Payment>} payment
- * @param {Amount[]} amounts
- * @returns {Promise<Payment[]>}
- */
```
Comment (erights) [inline packages/ERTP/src/types-ambient.js:189]: Getting rid of all this is a drive by. These declarations have not referred to anything for a very long time. Should have been deleted ages ago.

File: packages/ERTP/src/purse.js [hunk 2]
```diff
@@ -114,7 +126,8 @@ export const preparePurseKind = (
     {
       stateShape: {
         currentBalance: amountShape,
-        recoverySet: M.remotable('recoverySet'),
+        // NOTE: Schema change!
+        recoverySet: M.opt(M.remotable('recoverySet')),
```
Comment (erights) [inline packages/ERTP/src/purse.js:150]: Reviewers,

NOTE: Schema change!

What should we actually do about this?

File: packages/ERTP/src/paymentLedger.js [hunk 3]
```diff
@@ -156,18 +158,18 @@ export const preparePaymentLedger = (
    * - Every payment that is a key in the outer `paymentRecoverySets` weakMap is
    *   also in the recovery set indexed by that payment.
    * - Implied by the above but worth stating: the payment is only in at most one
-   *   recovery set.
+   *   one recovery set.
```
Comment (erights) [inline packages/ERTP/src/paymentLedger.js:161]: one one

File: packages/ERTP/src/paymentLedger.js [hunk 4]
```diff
@@ -156,18 +158,18 @@ export const preparePaymentLedger = (
...
+   * @type {WeakMapStore<Payment, SetStore<Payment>> | undefined}
    */
-  const paymentRecoverySets = provideDurableWeakMapStore(
-    issuerBaggage,
-    'paymentRecoverySets',
-  );
+  const paymentRecoverySets =
+    recoverySetOption === 'noRecoverySets'
+      ? undefined
+      : provideDurableWeakMapStore(issuerBaggage, 'paymentRecoverySets');
```
Comment (erights) [inline packages/ERTP/src/paymentLedger.js:174]: What if the revived issuer has `'noRecoverySets'` but the old state has recovery sets?

**Review by FUDCo — COMMENTED** [review]
This all looks quite plausible to me, but I don't think I'm qualified to comment on whether it's good to go, so I'll leave the approving part to somebody who knows this stuff better.

**Comment by erights** [comment]
Closing in favor of https://github.com/Agoric/agoric-sdk/pull/8418

**Pull request closed without merging**